### ISSUE TITLE
grpc: grpc-status in trailers does not trigger retries

### ISSUE THREAD

**Issue by storyicon** [envoyproxy/envoy#17629]
I noticed that currently for gRPC:

Envoy's retries will only detect `grpc-status` in `Headers`.When the server sets a custom Header, `grpc-status` will appear in the `Trailers`, which will cause Envoy's retries to not work:
https://github.com/grpc/grpc-go/blob/master/stream.go#L1369

I think we should do something in here:
https://github.com/envoyproxy/envoy/blob/main/source/common/router/router.cc#L1392-L1412

I noticed this comment, which makes me think that the developers didn't accidentally overlook the problem, but just put it aside for now:
https://github.com/envoyproxy/envoy/blob/main/source/common/router/router.cc#L1238-L1240

So are there any special considerations or difficulties? Will I get support if I submit a PR to fix this issue?

**Comment by ggreenway** [comment]
cc @alyssawilk

**Comment by mattklein123** [comment]
Yeah I agree with @alyssawilk, this will be very tricky, and I would rather not add this functionality to the router. I think a separate filter on the encoding path could potentially be used for this.

**Comment by Shikugawa** [comment]
The ALS retry process is currently only done when a connection cannot be established, but I think it would be useful to have a process to retry when Unavailable is listed in the Trailer. However, this feature needs to be implemented for this purpose, and we hope to implement it in the future.

**Comment by AurelPaulovic** [comment]
Is this related to #11049 ?
